### DISCUSSION TITLE
[Bug]: Waybar issue

### DISCUSSION THREAD

**Discussion by popglitchx** [JaKooLit/Hyprland-Dots#288] [Q&A]
### Description

Its lagging so much and
when i change workspace(ctrl + x) it takes a long time to update
some time nothing works
it has been working before like 3 months ago maybe an update broke it
### Steps to Reproduce
IDK

### Behavior
Lag and slow bar

### Screenshots

[[Video]](https://www.dropbox.com/scl/fi/h14vwtdf1lbg5fqne8xc2/2024-05-12-00-10-15.mkv?rlkey=he9zbt82ya0jqsftaom1tna29&st=bvz6hruy&dl=0)

### Environment
- Linux Distro: arch linux
- Linux Version: Linux thirteen 6.8.9-zen1-2-zen #1 ZEN SMP PREEMPT_DYNAMIC Tue, 07 May 2024 22:06:02 +0000 x86_64 GNU/Linux
- Hyprland Version : Tag: v0.40.0, commits: 4606
- Installation Environment: Main Hardware
- Other info: 1080p
- CPU: i5 2500
- GPU: AMD R7 360
- RAM: 13GB DDR3

**Comment by JaKooLit** [comment]
did you change something on waybar config? I cant emulate that error

**Comment by popglitchx** [comment]
No i did not i just installed hyprland
when i uesd your script is broke xorg for some reason

**Comment by JaKooLit** [comment]
how did you update? Ran Arch-Hyprland script or just pacman -Syu?

Next time, be more specific

First you show waybar is lagging

now you mentioned about xorg

**Comment by popglitchx** [comment]
I didn't update i just installed it 
I meant that i tried this dotfiles before and it was working fine

**Comment by JaKooLit** [comment]
as stated here

https://github.com/JaKooLit/Hyprland-Dots/wiki/Install_&_Update

"
 It is highly recommend, that in order to update to newer dot files (Hyprland-Dots), re-download and re-run the -Hyprland scripts above.

This is because to ensure that apps / packages are properly installed. You will get the latest released version of the dotfiles.
"

Dot files are evolving and some updates needs newer packages

**Comment by popglitchx** [comment]
I Just updated it and its the same still laging

**Comment by SherLock707** [comment]
> I Just updated it and its the same still laging

Open a terminal, `killall waybar` then launch Waybar with `waybar` and check if you see any error.

**Comment by popglitchx** [comment]
> killall waybar

I just did it and i got this 
```
[2024-05-12 09:37:06.731] [info] Using configuration file /home/lahcen/.config/waybar/config
[2024-05-12 09:37:06.731] [info] Including resource file: ~/.config/waybar/modules
[2024-05-12 09:37:06.736] [info] Discovered appearance 'dark'
[2024-05-12 09:37:06.736] [info] Using CSS file /home/lahcen/.config/waybar/style.css
[2024-05-12 09:37:06.745] [warning] Failed to open bar ipc connection: Socket path is empty
[2024-05-12 09:37:06.745] [info] Hyprland IPC starting
[2024-05-12 09:37:07.536] [info] Loading persistent workspaces from Waybar config
[2024-05-12 09:37:07.536] [info] Loading persistent workspaces from Hyprland workspace rules
[2024-05-12 09:37:08.338] [warning] module backlight/slider: Disabling module "backlight/slider", No backlight found
[2024-05-12 09:37:08.341] [warning] No batteries.
[2024-05-12 09:37:09.051] [info] Bar configured (width: 1920, height: 45) for output: HDMI-A-1

** (waybar:5236): WARNING **: 09:37:09.058: Status Notifier Item with bus name ':1.13' and object path '/org/ayatana/NotificationItem/nm_applet' is already registered
[2024-05-12 09:43:05.382] [warning] No workspace with id 3
```

**Comment by SherLock707** [comment]
> > killall waybar
> 
> I just did it and i got this 
> ```
> [2024-05-12 09:37:06.731] [info] Using configuration file /home/lahcen/.config/waybar/config
> [2024-05-12 09:37:06.731] [info] Including resource file: ~/.config/waybar/modules
> [2024-05-12 09:37:06.736] [info] Discovered appearance 'dark'
> [2024-05-12 09:37:06.736] [info] Using CSS file /home/lahcen/.config/waybar/style.css
> [2024-05-12 09:37:06.745] [warning] Failed to open bar ipc connection: Socket path is empty
> [2024-05-12 09:37:06.745] [info] Hyprland IPC starting
> [2024-05-12 09:37:07.536] [info] Loading persistent workspaces from Waybar config
> [2024-05-12 09:37:07.536] [info] Loading persistent workspaces from Hyprland workspace rules
> [2024-05-12 09:37:08.338] [warning] module backlight/slider: Disabling module "backlight/slider", No backlight found
> [2024-05-12 09:37:08.341] [warning] No batteries.
> [2024-05-12 09:37:09.051] [info] Bar configured (width: 1920, height: 45) for output: HDMI-A-1
> 
> ** (waybar:5236): WARNING **: 09:37:09.058: Status Notifier Item with bus name ':1.13' and object path '/org/ayatana/NotificationItem/nm_applet' is already registered
> [2024-05-12 09:43:05.382] [warning] No workspace with id 3
> ```

Likely something else taking up that socket and notification bus is running. Hunt it down through htop or something and kill it.

**Comment by popglitchx** [comment]
How i do that ?

**Comment by popglitchx** [comment]
Any one that can help me still having the same issue i re installed hyprland and the os and waybar

**Comment by SherLock707** [comment]
> Any one that can help me still having the same issue i re installed hyprland and the os and waybar

When you kill waybar and relaunch it in terminal, do you still see the socket error?

**Comment by popglitchx** [comment]
logs:
```
Hyprland-Dots ❯ waybar                                                              0.00s [main●] ~/Hyprland-Dots
[2024-05-15 07:09:49.656] [info] Using configuration file /home/lahcen/.config/waybar/config
[2024-05-15 07:09:49.657] [info] Including resource file: ~/.config/waybar/modules
[2024-05-15 07:09:49.660] [info] Discovered appearance 'dark'
[2024-05-15 07:09:49.660] [info] Using CSS file /home/lahcen/.config/waybar/style.css
[2024-05-15 07:09:49.670] [warning] Failed to open bar ipc connection: Socket path is empty
[2024-05-15 07:09:49.681] [info] Hyprland IPC starting
[2024-05-15 07:09:50.426] [info] Loading persistent workspaces from Waybar config
[2024-05-15 07:09:50.426] [info] Loading persistent workspaces from Hyprland workspace rules
[2024-05-15 07:09:50.623] [error] findCurController() failed: no bluetooth controller found
[2024-05-15 07:09:50.624] [info] Found device WFDZ Gaming Keyboard at '/dev/input/event2'

** (waybar:78940): WARNING **: 07:09:51.439: Status Notifier Item with bus name ':1.11' and object path '/org/ayatana/NotificationItem/nm_applet' is already registered
[2024-05-15 07:09:51.477] [info] Bar configured (width: 1904, height: 37) for output: HDMI-A-1
```

**Comment by SherLock707** [comment]
> logs:
> 
> ```
> Hyprland-Dots ❯ waybar                                                              0.00s [main●] ~/Hyprland-Dots
> [2024-05-15 07:09:49.656] [info] Using configuration file /home/lahcen/.config/waybar/config
> [2024-05-15 07:09:49.657] [info] Including resource file: ~/.config/waybar/modules
> [2024-05-15 07:09:49.660] [info] Discovered appearance 'dark'
> [2024-05-15 07:09:49.660] [info] Using CSS file /home/lahcen/.config/waybar/style.css
> [2024-05-15 07:09:49.670] [warning] Failed to open bar ipc connection: Socket path is empty
> [2024-05-15 07:09:49.681] [info] Hyprland IPC starting
> [2024-05-15 07:09:50.426] [info] Loading persistent workspaces from Waybar config
> [2024-05-15 07:09:50.426] [info] Loading persistent workspaces from Hyprland workspace rules
> [2024-05-15 07:09:50.623] [error] findCurController() failed: no bluetooth controller found
> [2024-05-15 07:09:50.624] [info] Found device WFDZ Gaming Keyboard at '/dev/input/event2'
> 
> ** (waybar:78940): WARNING **: 07:09:51.439: Status Notifier Item with bus name ':1.11' and object path '/org/ayatana/NotificationItem/nm_applet' is already registered
> [2024-05-15 07:09:51.477] [info] Bar configured (width: 1904, height: 37) for output: HDMI-A-1
> ```

Go in your currently active Waybar config and comment out all the module and keep just clock un-commented. then relaunch the Waybar. See if you still get the issue.

**Comment by popglitchx** [comment]
Then i could just use another bar config

**Comment by popglitchx** [comment]
can i use something like polybar or something

**Comment by ghost** [comment]
> can i use something like polybar or something

You can't. Polybar is for xOrg.

**Comment by SherLock707** [comment]
> Then i could just use another bar config

I suggested above method for debugging so that once you have Waybar running fine with single module, you can one-by-one start enabling module.

Also if you want to switch to new bar, I'd suggest Ags bar.

**Comment by popglitchx** [comment]
I didn't find Ags bar

**Comment by SherLock707** [comment]
Did you try my suggestion  with waybar?

**Comment by SherLock707** [comment]
>  I didn't find Ags bar

[https://github.com/Aylur/ags/tree/main/example/simple-bar](https://github.com/Aylur/ags/tree/main/example/simple-bar)

Copy and rename config and style in ags folder.


You can find more by searching 'ags' at [reddit](https://www.reddit.com/r/unixporn)

**Comment by JaKooLit** [comment]
Converted to discussion instead as it seems its not really a BUG
